iroh-gateway/src/core.rs

```diff
@@ -185,13 +190,36 @@ async fn get_handler(
     service_worker_check(&request_headers, cpath.to_string(), &state)?;
     unsuported_header_check(&request_headers, &state)?;
 
+    // check if cid is in the denylist
+    if state.bad_bits.read().await.is_bad(cid, cpath) {
```

Comment:
this adds overhead atm if not enabled, I would make bad_bits be an Option, so it is clear if it‘s enabled at all